File: 557-reverse-words-in-a-string-iii/557-reverse-words-in-a-string-iii.py
class Solution:
    def reverseWords(self, s: str) -> str:
        start=0
        end=0
        res=''
        lst= s.split(' ')
        for idx,i in enumerate(lst):
            res+=i[::-1]
            if idx!=len(lst)-1:
                res+=' '
        return res
                

File: 557-reverse-words-in-a-string-iii/557-reverse-words-in-a-string-iii.cpp
class Solution {
public:
    string reverseWords(string s) {
        string res= "", curr= "";
        for(char &c: s){
            if(c == ' '){
                reverse(curr.begin(), curr.end());
                res += curr + " ";
                curr = "";
            }else{
                curr += c;
            }
        }
        reverse(curr.begin(), curr.end());
        res += curr;
        return res;
    }
};